runtime/gc_vlhgc/ConfigurationIncrementalGenerational.cpp

```diff
@@ -272,8 +272,16 @@ MM_ConfigurationIncrementalGenerational::initialize(MM_EnvironmentBase *env)
 
 	bool result = MM_Configuration::initialize(env);
 
+	/* By default disable hot field depth copying */
+	env->disableHotFieldDepthCopy();
+
 	if (result) {
-		extensions->scavengerScanOrdering = MM_GCExtensions::OMR_GC_SCAVENGER_SCANORDERING_BREADTH_FIRST;
+		if (extensions->scavengerScanOrdering != MM_GCExtensions::OMR_GC_SCAVENGER_SCANORDERING_DYNAMIC_BREADTH_FIRST) {
+			extensions->scavengerScanOrdering = MM_GCExtensions::OMR_GC_SCAVENGER_SCANORDERING_BREADTH_FIRST;
+		} else {
+			extensions->adaptiveGcCountBetweenHotFieldSort = false;
+			extensions->hotFieldResettingEnabled = false;
```

Comment:
Do we really have to force disabling of these two flags - what bad can happen if somebody enables them through an option?

Reply:
Nothing bad, it would just be unnecessary checks for balanced as `adaptiveGcCountBetweenHotFieldSort` and `gcCountBetweenHotFieldReset` are dependant on `extensions->scavengerStats._gcCount` in `MM_HotFieldUtil::sortAllHotFieldData`.

Reply:
should we then add gcCount param to the method and pass it from the caller?

Reply:
Yes I could definitely do that and would just have to test the effects of `adaptiveGcCountBetweenHotFieldSort` for balanced. It proved to be effective for gencon so I have it enabled by default, however with 1000+ regions where different regions are selected for each pgc, this feature did not make sense to me in balanced and I was fairly certain it would not have a positive effect. `hotFieldResettingEnabled` does not need to be forced here as it is by default disabled, however for `adaptiveGcCountBetweenHotFieldSort`, should I either:

a) force it disabled for balanced (current implementation)
b) add gcCount param and disable `adaptiveGcCountBetweenHotFieldSort` by default (negative effect on gencon with dynamic breadth first)
c) enable it by default for gencon and balanced (I would need to test the effects of `adaptiveGcCountBetweenHotFieldSort` in balanced)

Reply:
Perhaps I do c) first and we take it from there?